designcompose/src/main/java/com/android/designcompose/LiveUpdateJni.kt

```diff
@@ -20,14 +20,28 @@ import androidx.annotation.VisibleForTesting
 import com.android.designcompose.serdegen.ConvertResponse
 import com.novi.bincode.BincodeDeserializer
 
+// HTTP Proxy configuration.
+internal data class HttpProxyConfig(val proxySpec: String)
+
+// Proxy configuration.
+//
+// Only HTTP proxy supported.
+internal class ProxyConfig {
+    var httpProxyConfig: HttpProxyConfig? = null
+}
+
 internal object LiveUpdateJni {
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    external fun jniFetchDoc(docId: String, requestJson: String): ByteArray
+    external fun jniFetchDoc(
+        docId: String,
+        requestJson: String,
+        proxyConfig: ProxyConfig
```

Comment:
Why not make this nullable?

Reply:
There was a crash with null JObjects on the Rust side with the get_ methods but not with say a null JString.  Resorted to doing this.

Reply:
:/

Ok, probably not that big a deal.